packages-internal/test-utils/src/createRenderer.tsx

```diff
@@ -306,7 +306,7 @@ function render(
       traceSync('forceUpdate', () =>
         testingLibraryRenderResult.rerender(
           React.cloneElement(element, {
-            'data-force-update': String(Math.random()),
+            ['data-force-update' as string]: String(Math.random()),
```

Comment:
here `element` type is `unknown`, since unknown doesn't have `data-force-update` type,  i've explicitly marked type as `as string`

Reply:
But if I understand correctly, this is casting the key, not the value 🤔 

What's the error thrown without the cast?

Reply:
But if I understand correctly, this is casting the key, not the value 🤔 

What's the error thrown without the cast?

Reply:
Instead of casting, i've tried to fix type error properly here =>https://github.com/mui/material-ui/pull/43402/commits/09f5298d8b63796516cb873a98ab1095eb29a295